include/gridtools/common/hugepage_alloc.hpp

```diff
@@ -11,25 +11,137 @@
 #pragma once
 
 #include <atomic>
+#include <cassert>
 #include <cstdlib>
+#include <fstream>
 #include <new>
+#include <stdexcept>
+#if __cpp_lib_int_pow2 >= 202002L
+#include <bit>
+#endif
+#ifdef __linux__
+#include <iostream>
+#include <regex>
+
+#include <sys/mman.h>
+#endif
 
 namespace gridtools {
+    namespace hugepage_alloc_impl_ {
+        inline std::size_t ilog2(std::size_t i) {
+#if __cpp_lib_int_pow2 >= 202002L
+            return std::bit_width(t) - 1;
+#else
+            std::size_t log = 0;
+            while (i >>= 1)
+                ++log;
+            return log;
+#endif
+        }
+
+        inline std::size_t cache_line_size() {
+            std::size_t value = 64; // default value for x86-64 archs
+#if __linux__
+            std::ifstream file("/sys/devices/system/cpu/cpu0/cache/index0/coherency_line_size");
```

Comment:
iostream library has significant compile time overhead.
I would suggest to use `read` from `unistd.h` and include the header conditionally only if `__linux__` is defined.

Reply:
Done.